### PR TITLE
Fixes reflector list expiration

### PR DESCRIPTION
**Pull request by yue9944882**
Fixes: https://github.com/kubernetes-client/java/issues/1533

### CODE REVIEW

**Comment by yue9944882** [comment]
/cc @tony-clarke-amdocs 

@brendandburns i suppose we should pick this back to 10 and 11 release, on which branch should open the cherry-picking pull?

**Comment by dsyer** [comment]
Master is beyond 11.0 now right (it has the breaking changes in the time API)? So we need to bump master to 12.0.0-SNAPSHOT and backport this change to 11.0 (at least).

**Comment by brendandburns** [comment]
@yue9944882 

The right branches are `release-10` and `release-11` 

Thanks!

File: util/src/test/java/io/kubernetes/client/informer/cache/ReflectorRunnableTest.java [hunk 1]
```diff
@@ -215,4 +217,42 @@ public V1PodList list(CallGeneratorParams params) throws ApiException {
...
   }
+
+  @Test
+  public void testReflectorListShouldHandleExpiredResourceVersion() throws ApiException {
+    String expectedResourceVersion = "100";
+    when(listerWatcher.list(any()))
+        .thenReturn(
+            new V1PodList().metadata(new V1ListMeta().resourceVersion(expectedResourceVersion)));
+    // constantly failing watches will make the reflector run only one time
+    when(listerWatcher.watch(any())).thenThrow(new ApiException(HttpURLConnection.HTTP_GONE, ""));
```
Comment (aaruna) [inline util/src/test/java/io/kubernetes/client/informer/cache/ReflectorRunnableTest.java:228]: What about the scenario wherein this API returns responseCode 200 but the body has an event that has status.code=410 ?
https://github.com/kubernetes-client/java/pull/1611/files#diff-31beacc81d066c2ab68de67a3587fee7bd7ef10d9ba42dc0a180b621751c6bffR226

**Pull request merged**